src/launchpad/artifact_processor.py

```diff
@@ -217,6 +256,22 @@ def _preprocess_artifact(
         else:
             return info
 
+    def _process_app_icon(
+        self,
+        organization_id: str,
+        project_id: str,
+        artifact_id: str,
+        artifact: Artifact,
+    ) -> str | None:
+        logger.info(f"Processing app icon for {artifact_id} (project: {project_id}, org: {organization_id})")
+        app_icon = artifact.get_app_icon()
+        if app_icon is None:
+            logger.info(f"No app icon found for {artifact_id} (project: {project_id}, org: {organization_id})")
+            return None
+
+        app_icon_id = self._objectstore_client.put(app_icon, compression="none")
```

Comment:
Left 'none' for now just for early testing, I think we can just use default once we're ready to productionize.